src/HydroComponent.cs

```diff
@@ -382,8 +451,10 @@ private async Task<string> RenderOnlineRootComponent(IPersistentState persistent
 
         PopulateDispatchers();
 
+        PopulateClientScripts();
+
         return !_skipOutput
-            ? await GenerateComponentHtml(componentId, persistentState)
+            ? await GenerateComponentHtml(componentId, persistentState, false)
```

Comment:
Why not execute scripts when rendering root component on Hydro request (isStatic = false), but execute on nested components (isStatic = true)?